src/GenAPI/Microsoft.DotNet.GenAPI/CSharpSyntaxWriter.cs

```diff
@@ -0,0 +1,113 @@
+            return AddPartialModifier(node);
+        }
+
+        public override SyntaxNode? VisitStructDeclaration(StructDeclarationSyntax node)
+        {
+            node = (StructDeclarationSyntax)base.VisitStructDeclaration(node)!;
+            return AddPartialModifier(node);
+        }
+
+        public override SyntaxNode? VisitMethodDeclaration(MethodDeclarationSyntax node)
+        {
+            // visit subtree first to normalize type names.
+            node = (MethodDeclarationSyntax)base.VisitMethodDeclaration(node)!;
+
+            if (node.Modifiers.Where(token => token.IsKind(SyntaxKind.AbstractKeyword)).Any())
+            {
+                return node;
+            }
+
+            if (node.ExpressionBody != null)
+            {
+                node = node.WithExpressionBody(null);
+            }
+
+            if (node.ReturnType.ToString() != "System.Void")
+            {
+                node = node.WithBody(GetThrowNullBody(true));
+            }
+            else
+            {
+                node = node.WithBody(GetEmptyBody(true));
+            }
+
+            return node.WithParameterList(node.ParameterList.WithTrailingTrivia(SyntaxFactory.Space));
+        }
+
+        public override SyntaxNode? VisitAccessorDeclaration(AccessorDeclarationSyntax node)
+        {
+            return node.Kind() switch
+            {
+
+                SyntaxKind.GetAccessorDeclaration => node.WithSemicolonToken(default)
+                                                         .WithKeyword(node.Keyword.WithTrailingTrivia(SyntaxFactory.Space))
+                                                         .WithBody(GetThrowNullBody(newLine: false)),
+                SyntaxKind.SetAccessorDeclaration => node.WithSemicolonToken(default)
+                                                         .WithKeyword(node.Keyword.WithTrailingTrivia(SyntaxFactory.Space))
+                                                         .WithBody(GetEmptyBody(newLine: false)),
+                _ => base.VisitAccessorDeclaration(node)
+            };
+        }
+
+        private SyntaxNode AddPartialModifier<T>(T node) where T: TypeDeclarationSyntax
+        {
+            return node.AddModifiers(SyntaxFactory.Token(SyntaxKind.PartialKeyword).WithTrailingTrivia(SyntaxFactory.Space));
+        }
+
+        private BlockSyntax GetEmptyBody(bool newLine = false)
+        {
+            BlockSyntax node = GetMethodBodyFromText(SyntaxFactory.Space.ToString(), newLine);
+            return node.WithOpenBraceToken(node.OpenBraceToken.WithTrailingTrivia(SyntaxFactory.Space));
+        }
+
+        private BlockSyntax GetThrowNullBody(bool newLine = false)
+        {
+            if (_exceptionMessage is string exceptionMessage)
+            {
+                return GetMethodBodyFromText(string.Format(" throw new PlatformNotSupportedException(\"{0}\"); ", exceptionMessage), newLine);
+            }
```

Comment:
No need to assign _exceptionMessage to a new variable. And use $"" which in this case is more readable.

```suggestion
            if (_exceptionMessage is not null)
            {
                return GetMethodBodyFromText($" throw new PlatformNotSupportedException(\"{_exceptionMessage}\"); "), newLine);
            }
```

Reply:
Thanks, nice formatting feature `$"{variable}"`